### ISSUE TITLE
Rewrite URLs in CSS files

### ISSUE THREAD

**Issue by neos-bot** [neos/flow-development-collection#520]
Jira issue originally created by user @robertlemke:

CSS files (and probably other types of assets) may contain relative or absolute URLs pointing to other resources – such as web fonts or images. Publishing theses assets as they are will result in broken links when using a CDN. Neos / Flow must replace the URLs to correct absolute URLs pointing to the CDN servers.

 Jira-URL: https://jira.neos.io/browse/FLOW-115

**Comment by sorenmalling** [comment]
@robertlemke 
I believe this a covered by todays build tools (webpack, grunt etc) and is not considered a nescessary core functionality of Flow.

Can I close this ticket?

**Comment by albe** [comment]
Yes that's manually doable with build tools, but maybe we can still improve the developer experience a bit? Maybe a documentation section on how to properly adjust URIs during/after the publishing process with those tools, while keeping generic/relative URIs in your assets?

**Comment by sorenmalling** [comment]
> Maybe a documentation section on how to properly adjust URIs during/after the publishing process with those tools

I'm all ear - do you mean the Flows own publishing mechanis or build tools such from "outside" Flow?

**Comment by bwaidelich** [comment]
Aren't resource URIs absolute anyways in Flow?

**Comment by sorenmalling** [comment]
> Aren't resource URIs absolute anyways in Flow?

If created with ex ResourceViewHelper and hence the ResourceManager, yes it's absolute

**Comment by albe** [comment]
Yes, but say you have a `css` file in your assets that contains an URI `background: url(/some/other/path/to/some/image.jpg)` and now you publish this `css` to a CDN via Flows `resource:publish` command, then the URI will no longer work (unless you end up with the exact same path to the `image.jpg` on that CDN, which is unlikely).

**Comment by sorenmalling** [comment]
This is more a topic of configuring your build tools and deployment setup, right? Can we call this a potential documentation topic? Maybe even a good documentation topic for Neos CMS integrators?
